src/Altinn.Platform/Altinn.Platform.Storage/Storage/Repository/TextRepository.cs

```diff
@@ -0,0 +1,128 @@
+
+            _client.CreateDocumentCollectionIfNotExistsAsync(
+                databaseUri,
+                documentCollection).GetAwaiter().GetResult();
+
+            _client.OpenAsync();
+        }
+
+        public async Task<TextResource> Get(string org, string app, string language)
+        {
+            try
+            {
+                string id = GetTextId(org, app, language);
+                Uri uri = UriFactory.CreateDocumentUri(databaseId, collectionId, id);
+                TextResource result = await _client
+                    .ReadDocumentAsync<TextResource>(
+                        uri,
+                        new RequestOptions { PartitionKey = new PartitionKey(org) });
+                return result;
+            }
+            catch (DocumentClientException dce)
+            {
+                if (dce.StatusCode == HttpStatusCode.NotFound)
+                {
+                    return null;
+                }
+
+                throw dce;
+            }
+        }
+
+        public async Task<TextResource> Create(string org, string app, TextResource textResource)
+        {
+            PreProcess(org, app, textResource);
+            ResourceResponse<Document> document = await _client.CreateDocumentAsync(_collectionUri, textResource);
+            TextResource result = JsonConvert.DeserializeObject<TextResource>(document.Resource.ToString());
+            return result;
+        }
+
+        public async Task<TextResource> Update(string org, string app, string language, TextResource textResource)
+        {
+            PreProcess(org, app, textResource);
+            Uri uri = UriFactory.CreateDocumentUri(databaseId, collectionId, textResource.Id);
+
+            ResourceResponse<Document> document = await _client
+                .ReplaceDocumentAsync(
+                    uri,
+                    textResource,
+                    new RequestOptions { PartitionKey = new PartitionKey(org) });
+
+            TextResource updatedResource = JsonConvert.DeserializeObject<TextResource>(document.Resource.ToString());
+            return updatedResource;
+        }
+
+        public async Task<bool> Delete(string org, string app, string language)
+        {
+            string id = GetTextId(org, app, language);
+            Uri uri = UriFactory.CreateDocumentUri(databaseId, collectionId, id);
+
+            ResourceResponse<Document> instance = await _client
+                .DeleteDocumentAsync(
+                    uri.ToString(),
+                    new RequestOptions { PartitionKey = new PartitionKey(org) });
+
+            return true;
+        }
+
+        private string GetTextId(string org, string app, string language)
+        {
+            return $"{org}-{app}-{language}";
+        }
+
+        /// <summary>
+        /// Pre processes the text resource. Creates id and adds partition key org
+        /// </summary>
+        private void PreProcess(string org, string app, TextResource textResource)
```

Comment:
should language be an input property so we ensure that it exists before using it?

Reply:
That could be an idea. This is checked at the controller level though.

Reply:
I'm thinking that we shouldn't have to rely on controller logic to verify that the data we receive is consistent. As the controller implementation might change at some point without us seeing the connection to the repository logic in the same PR

Reply:
Agreed. Added argument validation in repo as well, have a look :)